airflow/cli/cli_parser.py

```diff
@@ -60,10 +62,17 @@ def _check_value(self, action, value):
         if action.dest == 'subcommand' and value == 'celery':
             executor = conf.get('core', 'EXECUTOR')
             if executor not in (CELERY_EXECUTOR, CELERY_KUBERNETES_EXECUTOR):
-                message = (
-                    f'celery subcommand works only with CeleryExecutor, your current executor: {executor}'
-                )
-                raise ArgumentError(action, message)
+                executor_cls = import_string(ExecutorLoader.executors.get(executor, executor))
```

Comment:
I am not sure if this will allow the executor to load in all cases, and in particular if it works with executed provided by plugins.

Reply:
Should I maybe change that part to:
```suggestion
                executor_cls = ExecutorLoader.load_executor(ExecutorLoader.executors.get(executor, executor))
```

Reply:
@mik-laj is right. And loading executor might not be a good idea here actually, because there might be some side effects of instantiating the executor twice. But I think there is an easier way. Simply calling `ExecutorLoader.get_default_executor()` should return the executor object properly (as singleton as well - so it won't be initialized twice).

Reply:
@potiuk that's why I went with the import_string approach at first: `CeleryKubernetesExecutor`-based executors require celery_execytor and kubernetes_executor to be instantiated and passed to it at `__init__(self, celery_executor, kubernetes_executor)`, so the Celery-related command would still require Kubernetes dependencies installed. `ExecuterLoader.get_default_executor()` would still instantiate an executor object.

May I suggest splitting up methods in `ExecutorLoader` so that there would be a method for getting executor class by string name?